gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1179,43 +1179,69 @@ reconcile_when_close_toggled_cb (GtkToggleButton *togglebutton, ofx_info* info)
     info->run_reconcile = gtk_toggle_button_get_active (togglebutton);
 }
 
+static guint make_date_amount_key (time64 date, gnc_numeric amount)
+{
+    // Create a hash value that combines date and amount.
+    gchar buf[1024];
+    g_snprintf (buf, 1024, "%f%ld", gnc_numeric_to_double (amount), date);
```

Comment:
Don't use double, it can't reliably represent the numbers used in GnuCash. Do `"%ld/%ld:%ld", amount.num, amount.denom, date`.

1024 is a bit more buffer than you need, 64 will do nicely for 3 `int64_t`s, two punctuation marks, and the terminal `'\0'` and will fit in a cache row on most machines. 
Don't use the same magic number twice: say `g_snprintf (buf, size_of (buf), ...` instead. A magic-number pedant would suggest
```
#define HASH_BUFFER_SIZE 64
gchar buf[HASH_BUFFER_SIZE
#undef HASH_BUFFER_SIZE
```
but others might think that's overdoing it as long as the number occurs only in the decl of `buf`.